### PR TITLE
Update wasmparser to 0.45.0

### PR DESCRIPTION
**Pull request by abrown**
Adds many new operators and a few API changes.

- [x] This has not been discussed in any issue.
- [x] A short description of what this does, why it is needed: see title.
- [x] This PR contains test cases, if meaningful: uses existing tests.
- [x] A reviewer from the core maintainer team has been assigned for this PR.

### CODE REVIEW

File: cranelift-wasm/Cargo.toml [hunk 1]
```diff
@@ -26,8 +26,8 @@ target-lexicon = "0.9"
 
 [features]
 default = ["std", "basic-blocks"]
-std = ["cranelift-codegen/std", "cranelift-frontend/std", "wasmparser/std"]
-core = ["hashbrown", "cranelift-codegen/core", "cranelift-frontend/core", "wasmparser/core"]
+std = ["cranelift-codegen/std", "cranelift-frontend/std"]
+core = ["hashbrown", "cranelift-codegen/core", "cranelift-frontend/core"]
```
Comment (abrown) [inline cranelift-wasm/Cargo.toml]: I wasn't sure about removing these but it seemed that wasmparser no longer had these features?

File: cranelift-wasm/src/sections_translator.rs [hunk 2]
```diff
@@ -318,10 +319,7 @@ pub fn parse_element_section<'data>(
                 elems.into_boxed_slice(),
             )?
         } else {
-            return Err(wasm_unsupported!(
-                "unsupported passive elements section: {:?}",
-                kind
-            ));
+            return Err(wasm_unsupported!("unsupported passive elements section",));
```
Comment (abrown) [inline cranelift-wasm/src/sections_translator.rs]: `kind` is no longer `Debug`--should it be?

**Comment by abrown** [comment]
cc @yurydelendik

File: cranelift-wasm/Cargo.toml (hunk 1 above)
Comment (yurydelendik) [reply cranelift-wasm/Cargo.toml]: lgtm

File: cranelift-wasm/src/sections_translator.rs (hunk 2 above)
Comment (yurydelendik) [reply cranelift-wasm/src/sections_translator.rs]: maybe, but it is not useful here

**Pull request merged**